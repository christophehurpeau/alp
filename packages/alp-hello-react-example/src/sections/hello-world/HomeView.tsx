import { Stack, Typography, VStack } from "alouette";
import Head from "next/head";
import { appLogger } from "nightingale-app-console";
import { useState } from "react";
import type { ReactNode } from "react";
import { defineMessages, useIntl } from "react-intl";
import ClickToThrow from "./components/ClickToThrow";
import Counter from "./components/Counter";
import Hello from "./components/HelloComponent";
import { ToggleTheme } from "./components/ToggleTheme";

const logger = appLogger.child("HomePage");

function HelloWorld(): ReactNode {
  return <Typography>Hello World !</Typography>;
}

const messages = defineMessages({
  title: {
    id: "HomeView.title",
    defaultMessage: "Hello {name}!",
  },
});

export function HomeView(): ReactNode {
  const [name, setName] = useState((): string | undefined => {
    return "";
  });
  logger.info("render", { name });

  const handleChangeName = (newName: string): void => {
    logger.info("name changed", { newName });
    if (name === newName) return;
    setName(newName);
  };

  // useEffect(() => {
  // setSearchParams((queryParams) => {
  //   if (!name) {
  //     queryParams.delete('name');
  //   } else {
  //     queryParams.set('name', name);
  //   }
  // });

  // }, [name]);

  const intl = useIntl();
  return (
    <Stack padding="$4" $medium={{ padding: "$8" }}>
      <Head>
        <title>
          {intl.formatMessage(messages.title, { name: name || "World" })}
        </title>
      </Head>
      <VStack gap={1}>
        <ToggleTheme />
        <HelloWorld />
        <Hello name={name} onChangeName={handleChangeName} />
        <Counter />
        <ClickToThrow />
      </VStack>

      <div>
        <a href="/auth">Auth</a>
      </div>
    </Stack>
  );
}
