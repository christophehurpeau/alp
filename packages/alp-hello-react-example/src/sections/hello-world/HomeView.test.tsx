/**
 * @jest-environment jsdom
 */

import { act, fireEvent, render, screen } from '@testing-library/react';
import type { ReactNode } from 'react';
import { IntlProvider } from 'react-intl';
import { TamaguiProvider } from 'tamagui';
import TamaguiConfig from '../../tamagui.config';
import { HomeView } from './HomeView';

function Providers({ children }: { children: ReactNode }): ReactNode {
  return (
    <TamaguiProvider config={TamaguiConfig}>
      <IntlProvider locale="en" defaultLocale="en">
        {children}
      </IntlProvider>
    </TamaguiProvider>
  );
}

describe('HomeView', () => {
  test('hello without name', () => {
    render(<HomeView />, { wrapper: Providers });
    expect(screen.getByTestId('hello-text')).toMatchSnapshot();
  });

  test('hello with name', () => {
    // TODO pass query params: name=John
    render(<HomeView />, { wrapper: Providers });

    // expect(getByTestId('hello-text')).toMatchInlineSnapshot(`
    //   <span
    //     class="h199brc1"
    //     data-testid="hello-text"
    //   >
    //     Hello {name}! ; {"name":"John"}
    //   </span>
    // `);

    // https://github.com/tamagui/tamagui/issues/1329
    const input = screen.getByLabelText('Input your name');
    act(() => {
      fireEvent.change(input, { target: { value: 'Jane' } });
    });
    expect(screen.getByTestId('hello-text')).toMatchSnapshot();
  });
});
