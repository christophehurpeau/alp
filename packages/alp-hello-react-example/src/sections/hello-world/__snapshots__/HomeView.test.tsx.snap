// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView hello with name 1`] = `
<span
  class="font_unset _display-inline _boxSizing-border-box _wordWrap-break-word _marginTop-0px _marginRight-0px _marginBottom-0px _marginLeft-0px _color-675002279 _fontSize-229441158"
  data-testid="hello-text"
>
  Hello Jane!
</span>
`;

exports[`HomeView hello without name 1`] = `
<span
  class="font_unset _display-inline _boxSizing-border-box _wordWrap-break-word _marginTop-0px _marginRight-0px _marginBottom-0px _marginLeft-0px _color-675002279 _fontSize-229441158"
  data-testid="hello-text"
>
  Hello World!
</span>
`;
